### DISCUSSION TITLE
Error on Update

### DISCUSSION THREAD

**Discussion by kaelsonofkrypto** [nagadomi/nunif#109] [General]
Hi, a friend of mine is having issues installing the update to iw3 and now it doesnt want to work for her, i had her try to do update.bat but it didnt work. Then i had here try to do a fresh install but it still didnt work, 

Here is the screenshot of error during install

<img width="677" alt="Error during installv2" src="https://github.com/nagadomi/nunif/assets/164638883/dd9f37a6-7a6c-4e6e-b13b-8ecdc563c76c">

After she installed and tried to convert she got this error message in iw3

<img width="359" alt="Error message in iw3" src="https://github.com/nagadomi/nunif/assets/164638883/3e75db95-212f-4cae-b508-d2953e17dd98">

and i then had her run the nunif prompt and this is the log 

<img width="1280" alt="nunif command promptv2" src="https://github.com/nagadomi/nunif/assets/164638883/4cc922bb-be27-4d2f-bdcd-e0d1af8a48d2">

**Comment by nagadomi** [comment]
It is a network error.
internet connection is unstable or access to download.pytorch.org is blocked by a firewall, or some download size limit, etc.

I suggest try again later.
If the error occurs repeatedly, you can zip the pre-setup folder and send it to the other person. The file size is large, but it is portable.

**Reply by nagadomi** [reply]
If network env requires HTTP Proxy, it is not currently supported.

**Comment by kaelsonofkrypto** [comment]
Thanks for all your help. I just let her know what you mentioned and she said that she doesn't have a Firewall enabled. It was working for her previously before it didn't start acting up for her until after she did the last update. so I told her to just restart her PC to see if it will work because we had already tried to do a reinstall and it didn't work. It was late so she's gonna get some rest but I will confirm with her tomorrow to find out if the restart worked for her and I will also tell her to unplug her modem and plug it back in to see if that will help too.

**Comment by kaelsonofkrypto** [comment]
hi i wanted to provide an update, so we still have not been able to get it to work for her, it was working for her before update but still doesnt want to work, we tried enabling remote on her firewall adding python still didnt work so i had her change her settings back to what she had before,  we had her restart pc and still cant get it to work for her, another guy i chat with is also saying he is now having the issue to after the update, he just posted this today

![image](https://github.com/nagadomi/nunif/assets/164638883/ddd264cc-1f32-4454-bff1-eb8ad4b7cd15)

**Comment by Ryudoadema** [comment]
Indeed. Happening to me too. 3 fresh installs trying different things and can't figure it out.

**Comment by nagadomi** [comment]
This problem does not occur to all of us, so I believe it is some kind of environment-dependent problem.

Here are some checks I can think of
- Make sure you have 64-bit Windows 7 or higher.
- Apply all Windows Updates and reboot (this may resolve SSL related issues)
- Make sure you are not in unstable network env (Mobile Network, Free Wi-Fi, Great Fire Wall, Tor, VPN, etc)
- Make sure https://download.pytorch.org/whl/torch/ can be browsed successfully
- Make sure you have enough disk space
- 
Error Details:
PyTorch was recently updated(2024-03-28) and the installer is trying to download it and failing due to some problem.
Also, `download.pytorch.org` uses CloudFront(Content Delivery Network), different servers are used depending on the user's location. It is possible that there is a problem on the server side.

**Reply by nagadomi** [reply]
The following issue says that Kaspersky Internet Security was the reason.
https://github.com/pypa/pip/issues/12406

It is possible that some security software is blocking the internet connection from update.bat or python.exe.

**Reply by kaelsonofkrypto** [reply]
She told me that she didn't have any antivirus software. I signed in to remote desktop to look at her computer, and I didn't see kaspersky Internet on her computer.  We also went into network settings, and I could see windows defender was the only thing she had. If Windows defender is blocking it for some reason, what setting would we have to change? The strange thing about this, though, is I literally saw it live with her that it was working before the update. I had her do update installer, update bat, I even had her delete that whole folder and then redownloaded and re extracted and tried to do a clean install and it didn't work.

**Reply by nagadomi** [reply]
I am not a regular Windows user, I don't know much about Windows.
You should be able to search for Windows Defender settings on Google.

----

If it has worked in the past, I recommend trying to install nunif-windows-package by following [README instruction](https://github.com/nagadomi/nunif/blob/master/windows_package/docs/README.md) (especially Security > Unblock) with a different folder path.
I don't know how the Firewall identifies the program, but if it identifies it by file path, installing to a different folder should help avoid it. If it identifies the file by its hash, then there is only way to change the setting.

**Reply by Ryudoadema** [reply]
One hint for me is that my iw3 install works with ZoeD_Any_N, but not Any_S or Any_B. So that should help get to the root cause of the issue. When I try to convert with the Any_N or Any_S it gives a "winerror 10054 existing connection forcibly closed by the remote host".

**Reply by nagadomi** [reply]
In case there is no error during installation, but an error occurs when changing the model,
the most likely reason is that the internet connection is allowed from `python.exe`, but not from `pythonw.exe`.
At installation, `ZoeD_N` is downloaded with `python.exe -m iw.download_models` command and GUI is launched with `pythonw.exe -m iw3.gui` command.

This guess can be verified in the following ways
1. Run `nunif-prompt.bat`
1. Type `python -m iw3.gui` and press Enter
![screen](https://github.com/nagadomi/nunif/assets/287255/6f0781ec-de59-4391-8a0c-d21f85e38e16)

If there is no error in the launched GUI, the cause is that `pythonw.exe` is blocked by firewall.

----
One other possible reason is that there is a problem with the connection to `huggingface.co`.
The ZoeDepth models are downloaded from `github.com` and the DepthAnything models are downloaded from `huggingface.co`.

**Reply by Ryudoadema** [reply]
Thanks, I'll look into those

**Reply by nagadomi** [reply]
If the problem is only with downloading large model files, you can also download it manually.

https://huggingface.co/spaces/LiheYoung/Depth-Anything/tree/main/checkpoints
https://huggingface.co/spaces/LiheYoung/Depth-Anything/tree/main/checkpoints_metric_depth

```
depth_anything_metric_depth_indoor.pt
depth_anything_metric_depth_outdoor.pt
depth_anything_vitl14.pth
depth_anything_vitb14.pth
depth_anything_vits14.pth
```
Place the above five files in`nunif\iw3\pretrained_models\hub\checkpoints`.

**Reply by Ryudoadema** [reply]
Perfect, it's working! I thought that might be the issue, but wasn't exactly sure the source or destination of those.
